devtools/replica-setup/start_replication.sh

```diff
@@ -18,14 +18,15 @@ else
 fi
 
 # Use the EXECUTED_GTID_SET variable from the previous steps
-if [ ! -z "$EXECUTED_GTID_SET" ]; then
+if [ $LOG_POS_MODE == "OFF" ] && [ ! -z "$EXECUTED_GTID_SET" ]; then
   mysqlsh --sql --host=${MYDUCK_HOST} --port=${MYDUCK_PORT} --user=root --no-password <<EOF
 SET GLOBAL gtid_purged = "${EXECUTED_GTID_SET}";
 EOF
 fi
 
 # Connect to MySQL and execute the replication configuration commands
-mysqlsh --sql --host=${MYDUCK_HOST} --port=${MYDUCK_PORT} --user=root --no-password <<EOF
+if [ $LOG_POS_MODE == "OFF" ]; then
```

Comment:
It's better to combine the duplicated part of `CHANGE REPLICA SOURCE TO`: 
```bash
CHANGE_REPLICATION_CMD="CHANGE REPLICATION SOURCE TO
  SOURCE_HOST='${MYSQL_HOST_FOR_REPLICA}',
  SOURCE_PORT=${MYSQL_PORT},
  SOURCE_USER='${MYSQL_USER}',
  SOURCE_PASSWORD='${MYSQL_PASSWORD}'"

if [ "$LOG_POS_MODE" == "OFF" ]; then
  mysqlsh --sql --host=${MYDUCK_HOST} --port=${MYDUCK_PORT} --user=root --no-password <<EOF
$CHANGE_REPLICATION_CMD
;
START REPLICA;
EOF
else
  mysqlsh --sql --host=${MYDUCK_HOST} --port=${MYDUCK_PORT} --user=root --no-password <<EOF
$CHANGE_REPLICATION_CMD,
  SOURCE_LOG_FILE='${BINLOG_FILE}',
  SOURCE_LOG_POS=${BINLOG_POS}
;
START REPLICA;
EOF
fi
```

Reply:
fixed